gms/gossiper.cc

```diff
@@ -1380,10 +1380,14 @@ endpoint_state& gossiper::get_endpoint_state(inet_address ep) {
 }
 
 future<> gossiper::reset_endpoint_state_map() {
```

Comment:
It's a public interface so I it's not very elegant to assume it will be called only on shard 0.
But `invoke_on` on the same shard is inexpensive.  It doesn't actually perform a cross-shard task.